Classes/Service/NewsOverwrite.php

```diff
@@ -59,25 +54,24 @@ public function overWriteNewsPropertiesByIndexObject(News $news, Index $index) {
 	}
 
 	/**
-	 * Get combined time
+	 * Get combined time as datetime
 	 *
-	 * - Remove the time of the date information
-	 * - Remove the date if the time information
 	 * - combine both values
 	 *
 	 * @param int $dateTimestamp
 	 * @param int $timeTimestamp
 	 *
-	 * @return int
+	 * @return /DateTime
 	 */
-	protected function getCombinedTime($dateTimestamp, $timeTimestamp) {
-		$dateTimestamp = floor($dateTimestamp / self::DAY) * self::DAY;
-		if ($timeTimestamp <= 0) {
-			return $dateTimestamp;
-		}
-		if ($timeTimestamp < self::DAY) {
-			return $dateTimestamp + $timeTimestamp;
-		}
-		return $dateTimestamp + ($timeTimestamp % self::DAY);
+	protected function getCombinedTimeAsDatetime($dateTimestamp, $timeTimestamp) {
+
+		$newDateTime = new \DateTime('@' .$dateTimestamp);
```

Comment:
For the next time: https://github.com/lochmueller/calendarize/blob/master/Classes/Utility/DateTimeUtility.php#L140